yigitcanTask/force-app/main/default/classes/accountDetailsController.cls

```diff
@@ -0,0 +1,103 @@
+public class accountDetailsController {
+
+    public static final Integer ACCOUNT_NAME_MAX_LENGTH = 100;
+    public static final String ACCOUNT_LENGTH_ERROR_MESSAGE = 'Length is less than '+ACCOUNT_NAME_MAX_LENGTH+' characters';
+    
+    public List <Contact> contacts {get;set;}
+    public String searchKey {get;set;}
+    public Account account {get;set;}
+
+    public accountDetailsController(ApexPages.StandardController accountDetailsController ) {
+        this.account = (Account)accountDetailsController.getRecord();
+        this.contacts = this.getAllContacts();
+    }
+
+    public void primaryToContact() {
+        try {
+            String passedParam1 = Apexpages.currentPage().getParameters().get('contactsParam');
+            
+            List<Contact> allAccountContacts = this.getAllContacts();
+            for(Contact c : allAccountContacts) {
```

Comment:
try to avoid variables are only 1 letter long.
for this specific case please rename "c" to "contactRecord"